core/src/stages/compute_ik.cpp

```diff
@@ -336,23 +341,25 @@ void ComputeIK::compute() {
 		marker.color.a *= 0.5;
 		eef_markers.push_back(marker);
 	};
-	const auto& links_to_visualize = moveit::core::RobotModel::getRigidlyConnectedParentLinkModel(link)
-	                                     ->getParentJointModel()
-	                                     ->getDescendantLinkModels();
-	if (colliding) {
-		SubTrajectory solution;
-		std::copy(frame_markers.begin(), frame_markers.end(), std::back_inserter(solution.markers()));
-		generateCollisionMarkers(sandbox_state, appender, links_to_visualize);
-		std::copy(eef_markers.begin(), eef_markers.end(), std::back_inserter(solution.markers()));
-		solution.markAsFailure();
-		// TODO: visualize collisions
-		solution.setComment(s.comment() + " eef in collision: " + listCollisionPairs(collisions.contacts, ", "));
-		auto colliding_scene{ scene->diff() };
-		colliding_scene->setCurrentState(sandbox_state);
-		spawn(InterfaceState(colliding_scene), std::move(solution));
-		return;
-	} else
-		generateVisualMarkers(sandbox_state, appender, links_to_visualize);
+	auto const& connected_parent_link_model = moveit::core::RobotModel::getRigidlyConnectedParentLinkModel(link);
+	if (connected_parent_link_model && connected_parent_link_model->getParentJointModel()) {
+		const auto& links_to_visualize = connected_parent_link_model->getParentJointModel()->getDescendantLinkModels();
```

Comment:
Again, under which circumstances do you run into a segfault here?
`getRigidlyConnectedParentLinkModel()` should always return a valid LinkModel.

Reply:
You're right, if link is guaranteed not to be a nullptr, `getRigidlyConnectedParentLinkModel` doesn't return a nullptr :+1: